lldb/source/Plugins/SymbolFile/DWARF/DWARFIndex.cpp

```diff
@@ -126,3 +126,62 @@ bool DWARFIndex::GetFullyQualifiedTypeImpl(
     return callback(die);
   return true;
 }
+
+void DWARFIndex::GetNamespacesWithParents(
+    ConstString name, llvm::ArrayRef<llvm::StringRef> parent_names,
```

Comment:
Instead of just an array of names, we should try to use the `llvm::ArrayRef<CompilerContext>`. The .debug_names table has the DW_TAG in each entry and we can further reduce the number of things and possibly not even need to check the name if the DW_TAG_ doesn't match of the CompilerContext type isn't set to Any.